includes/Modules/Analytics_4.php

```diff
@@ -311,8 +324,9 @@ function ( $audience ) {
 			'pre_update_option_googlesitekit_analytics-4_settings',
 			function ( $new_value, $old_value ) {
 				if ( $new_value['propertyID'] !== $old_value['propertyID'] ) {
-					$new_value['availableCustomDimensions'] = null;
-					$new_value['availableAudiences']        = null;
+					$new_value['availableCustomDimensions']            = null;
+					$new_value['availableAudiences']                   = null;
+					$new_value['audienceSegmentationSetupCompletedBy'] = null;
```

Comment:
Hi @nfmohit, thanks for applying this fix.

It does strike me that we could have taken an alternative route of overriding the `Analytics_4::Settings::merge()` method, to allow `null` to be merged for these properties. This would allow us to use `merge()` as we want to and keep the audience-related reset logic encapsulated within `Reset_Audiences` as originally intended.

Could I please as you to create a followup issue as an Audience Segmentation P2 where we can explore this further?

I'll go ahead and merge this for now. Thanks!

Reply:
Absolutely, I've created #9388 as a follow-up, thanks Tom!